bitwarden_access_manager.py

```diff
@@ -0,0 +1,678 @@
+                    for x in data["member_access"]
+                )
+            )
+
+        return Collection(
+            id=data["collection_id"],
+            external_id=data["external_id"],
+            group_access=group_access,
+            member_access=member_access,
+        )
+
+    def format_toml(self) -> str:
+        result = (
+            "[[collection]]\n"
+            f"collection_id = {self.id}\n"
+            f'external_id = "{self.external_id}"\n'
+        )
+
+        if self.member_access is not None:
+            member_access_lines = [
+                "  " + a.format_toml() for a in sorted(self.member_access)
+            ]
+            if len(member_access_lines) > 0:
+                result = (
+                    result
+                    + "member_access = [\n"
+                    + ",\n".join(member_access_lines)
+                    + ",\n]\n"
+                )
+            else:
+                result = result + "member_access = []\n"
+
+        if self.group_access is not None:
+            group_access_lines = [
+                "  " + a.format_toml() for a in sorted(self.group_access)
+            ]
+            if len(group_access_lines) > 0:
+                result = (
+                    result
+                    + "group_access = [\n"
+                    + ",\n".join(group_access_lines)
+                    + ",\n]"
+                )
+            else:
+                result = result + "group_access = []"
+        return result
+
+
+class BitwardenClient(NamedTuple):
+    connection: HTTPSConnection
+    bearer_token: str
+
+    @staticmethod
+    def new(client_id: str, client_secret: str) -> BitwardenClient:
+        response = requests.post(
+            "https://identity.bitwarden.com/connect/token",
+            headers={"Content-Type": "application/x-www-form-urlencoded"},
+            data={
+                "grant_type": "client_credentials",
+                "scope": "api.organization",
+                "Accept": "application/json",
+            },
+            auth=(client_id, client_secret),
+        )
+        bearer_token = response.json()["access_token"]
+        return BitwardenClient(HTTPSConnection("api.bitwarden.com"), bearer_token)
+
+    def _http_get(self, url: str) -> HTTPResponse:
+        self.connection.request(
+            method="GET",
+            url=url,
+            headers={
+                "Accept": "application/json",
+                "Authorization": f"Bearer {self.bearer_token}",
+            },
+        )
+
+        return self.connection.getresponse()
+
+    def get_groups(self) -> Iterable[Group]:
+        groups = json.load(self._http_get(f"/public/groups"))
+        for group in groups["data"]:
+            yield Group(
+                id=group["id"],
+                name=group["name"],
+                access_all=group["accessAll"],
+            )
+
+    def get_collection_members(
+        self, groups: Any, org_members: Dict[str, Member]
```

Comment:
```suggestion
        self, groups: List[Dict[str, Any]], org_members: Dict[str, Member]
```
We can be a bit more precise with the type here.

Reply:
Added in https://github.com/ChorusOne/github-access-manager/pull/10/commits/d8013aa4cd1cccbea166a1cda7c76664732efb2d